apis/python/src/tiledbsoma/_read_iters.py

```diff
@@ -261,19 +268,11 @@ class BlockwiseTableReadIter(BlockwiseReadIterBase[BlockwiseTableReadIterResult]
 
     def _create_reader(self) -> Iterator[BlockwiseTableReadIterResult]:
         """Private. Blockwise Arrow Table iterator, restricted to a single axis"""
-        if self.eager and not self.eager_iterator_pool:
-            with ThreadPoolExecutor() as _pool:
-                yield from (
-                    self._reindexed_table_reader(_pool)
-                    if self.axes_to_reindex
-                    else self._table_reader()
-                )
-        else:
-            yield from (
-                self._reindexed_table_reader(_pool=self.eager_iterator_pool)
-                if self.axes_to_reindex
-                else self._table_reader()
-            )
+        yield from (
+            self._reindexed_table_reader(_pool=self._threadpool)
+            if self.axes_to_reindex
+            else self._table_reader()
+        )
```

Comment:
Since this no longer uses a context manager, there’s no need to `yield from` here; you can just `return` the generator.